### DISCUSSION TITLE
Callback on a StandardListViewItem?

### DISCUSSION THREAD

**Discussion by Dirleye** [slint-ui/slint#4963] [Q&A] [answered]
Hi there, I'd like to return the text in a StandardListViewItem when said item is clicked.
I appreciate that this probably requires some custom ListViewItem component but I'm not sure how to implement.

Many thanks for any suggestions, though I'm still getting used to it this is a very nice toolkit to work with. :smile:

**Comment by ogoffart** [comment, marked as answer]
There is a `item-pointer-event` callback in StandardListView which you can use for that: https://slint.dev/releases/1.5.1/docs/slint/src/language/widgets/standardlistview#callbacks


Example:
```slint
import { StandardListView } from "std-widgets.slint";
export component Example inherits Window {
    width: 150px;
    height: 150px;
    StandardListView {
        width: 150px;
        height: 150px;
        model: [ { text: "Blue"}, { text: "Red" }, { text: "Green" },
            { text: "Yellow" }, { text: "Black"}, { text: "White"},
            { text: "Magenta" }, { text: "Cyan" },
        ];
        item-pointer-event(i, ev) => {
            if ev.button == PointerEventButton.left && ev.kind == PointerEventKind.up {
                debug(self.model[i].text);
            }
        }
    }
}
```

**Reply by Dirleye** [reply]
Excellent, thank you very much!
